.github/workflows/publish.yaml

```
@@ -0,0 +1,31 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE for licensing details.
+
+name: Release to latest/edge
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  ci-tests:
+    uses: ./.github/workflows/ci.yaml
+
+  release-to-charmhub:
+    name: Release cephfs-client to Charmhub
+    needs:
+      - ci-tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Select charmhub channel
+        uses: canonical/charming-actions/channel@2.4.0
+        id: channel
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.4.0
+        with:
+          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          channel: "${{ steps.channel.outputs.name }}"
```

[nit]: Missing newline at the end of the file:

```suggestion
          channel: "${{ steps.channel.outputs.name }}"
          
```